paparazzi/paparazzi-gradle-plugin/build.gradle

```diff
@@ -31,7 +31,7 @@ sourceSets {
 def generateVersion = tasks.register("pluginVersion") {
   def outputDir = file('src/generated/kotlin')
 
-  inputs.property 'version', version
+  inputs.property 'version', project.property("VERSION_NAME")
```

Comment:
This looks fishy. This could mean that the POM will have missing metadata, if the `project.version` is not set up correctly. I would keep these as `version`, and set `project.version = project.property("VERSION_NAME")` at the top somewhere.

Reply:
The property is hardcoded in the committed gradle.properties file and `project.property` will throw an exception if the property does not exist. The config here will not in any way determine what ends up in the pom, this is handled by the publish plugin which does this based on the same property and has been using this property for that purpose since the beginning of its existence

Reply:
> I needed to adjust the code that injects the plugin version into the code because the publish plugin does not send project.version anymore.

so, earlier project.version was set, which was propagated to many places, now you're saying it's not done. `project.version` is the central place to define the version of the module, its artifacts, metadata, components, publications etc.

This `inputs.property` is only used for doing up-to-date checks (and cache key generation), it won't actually be an input. The one lower down will generate the Kotlin code so that **at runtime** consumers can see which version is there.

I did a quick diff and the version is actually correctly present on the branch, but many other things are wrong, will report them separately.

Reply:
>  so, earlier project.version was set, which was propagated to many places, now you're saying it's not done. 

Yes.

> project.version is the central place to define the version of the module, its artifacts, metadata, components, publications etc.

It's the default used for publications but the publish plugin is giving higher priority to the `VERSION_NAME` property if that is set and that is how paparazzi was already setting the version before.

Reply:
I see, so now the Paparazzi code-gen is matching the publishing plugin's behavior, which is different than Gradle's, but at least it's consistent within this build.